mm2src/coins/lightning/ln_platform.rs

```diff
@@ -0,0 +1,688 @@
+            }
+            channel_manager.transactions_confirmed(
+                &confirmed_transaction_info.header,
+                &[(
+                    confirmed_transaction_info.index,
+                    &confirmed_transaction_info.transaction,
+                )],
+                confirmed_transaction_info.height,
+            );
+            chain_monitor.transactions_confirmed(
+                &confirmed_transaction_info.header,
+                &[(
+                    confirmed_transaction_info.index,
+                    &confirmed_transaction_info.transaction,
+                )],
+                confirmed_transaction_info.height,
+            );
+        }
+    }
+
+    pub async fn get_channel_closing_tx(&self, channel_details: SqlChannelDetails) -> SaveChannelClosingResult<String> {
+        let from_block = channel_details
+            .funding_generated_in_block
+            .ok_or_else(|| MmError::new(SaveChannelClosingError::BlockHeightNull))?;
+
+        let tx_id = channel_details
+            .funding_tx
+            .ok_or_else(|| MmError::new(SaveChannelClosingError::FundingTxNull))?;
+
+        let tx_hash =
+            H256Json::from_str(&tx_id).map_to_mm(|e| SaveChannelClosingError::FundingTxParseError(e.to_string()))?;
+
+        let funding_tx_bytes = loop {
+            match self
+                .coin
+                .as_ref()
+                .rpc_client
+                .get_transaction_bytes(&tx_hash)
+                .compat()
+                .await
+            {
+                Ok(tx) => break tx,
+                Err(e) => {
+                    log::error!("Error getting funding tx bytes: {}", e);
+                    Timer::sleep(GET_FUNDING_TX_INTERVAL).await;
+                },
+            }
+        };
+
+        let closing_tx = self
+            .coin
+            .wait_for_tx_spend(
+                &funding_tx_bytes.into_vec(),
+                (now_ms() / 1000) + 3600,
+                from_block,
+                &None,
+            )
+            .compat()
+            .await
+            .map_to_mm(SaveChannelClosingError::WaitForFundingTxSpendError)?;
+
+        let closing_tx_hash = format!("{:02x}", closing_tx.tx_hash());
+
+        Ok(closing_tx_hash)
+    }
+}
+
+impl FeeEstimator for Platform {
+    // Gets estimated satoshis of fee required per 1000 Weight-Units.
+    fn get_est_sat_per_1000_weight(&self, confirmation_target: ConfirmationTarget) -> u32 {
+        let platform_coin = &self.coin;
+
+        let default_fee = match confirmation_target {
+            ConfirmationTarget::Background => self.default_fees_and_confirmations.background.default_feerate,
+            ConfirmationTarget::Normal => self.default_fees_and_confirmations.normal.default_feerate,
+            ConfirmationTarget::HighPriority => self.default_fees_and_confirmations.high_priority.default_feerate,
+        } * 4;
```

Comment:
I think I asked it already but forgot 😅 
What does `4` mean? How about adding a constant?

Reply:
😄 Well It can be removed, and the configuration `default_feerate` can be renamed `default_fee_per_kb` so the transaction size is used instead of the [weight](https://github.com/lightning/bolts/blob/master/03-transactions.md#expected-weight-of-the-commitment-transaction) in config, and the division by 4 here will get the right sats per 1000 weights. https://github.com/KomodoPlatform/atomicDEX-API/blob/152b2afafa5d8926aa7bd561e2ec433786b02600/mm2src/coins/lightning/ln_platform.rs#L608
A good explanation of transaction weight is [here](https://learnmeabitcoin.com/technical/transaction-weight), it's related to how many transactions can a block include post segwit and it's approximately 4x how many transactions can be included in pre segwit blocks.